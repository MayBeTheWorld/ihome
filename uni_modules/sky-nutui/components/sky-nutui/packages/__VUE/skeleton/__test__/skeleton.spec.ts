import { mount, config } from '@vue/test-utils';
import Skeleton from '../index.vue';

import NutAvatar from '../../avatar/index.vue';
import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutAvatar,
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('should render with row width array correctly', () => {
  const wrapper = mount(Skeleton, {
    props: {
      width: '100px'
    }
  });
  const skeleton: any = wrapper.find('.blockClass');
  expect(skeleton.element.style.width).toBe('100px');
});

test('should allow to disable animation', async () => {
  const wrapper = mount(Skeleton, {
    props: {
      row: '1'
    }
  });

  expect(wrapper.find('.skeleton').exists()).toBeTruthy();

  await wrapper.setProps({ animated: false });
  expect(wrapper.find('.skeleton-animation').exists()).toBeFalsy();
});

test('should change avatar size when using avatarSize prop', () => {
  const wrapper = mount(Skeleton, {
    props: {
      avatar: true,
      avatarSize: '20px'
    }
  });

  const avatar: any = wrapper.find('.avatarClass');
  expect(avatar.element.style.width).toMatchSnapshot('20px');
  expect(avatar.element.style.height).toMatchSnapshot('20px');
});

test('should change avatar shape when using avatarShape prop', () => {
  const wrapper = mount(Skeleton, {
    props: {
      avatar: true,
      avatarShape: 'square'
    }
  });
  expect(wrapper.find('.avatarClass').html()).toMatchSnapshot();
});

test('should be round when using round prop', () => {
  const wrapper = mount(Skeleton, {
    props: {
      title: true,
      round: true,
      avatar: true
    }
  });
  expect(wrapper.find('.avatarClass').exists()).toBeTruthy();
});

test('should render default slot', () => {
  const wrapper = mount(Skeleton, {
    slots: {
      default: 'default'
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});
