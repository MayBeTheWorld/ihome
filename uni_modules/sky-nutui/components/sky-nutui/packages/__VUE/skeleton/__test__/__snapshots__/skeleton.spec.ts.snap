// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change avatar shape when using avatarShape prop 1`] = `
"<view style=\\"width: 50px; height: 50px; background-color: rgb(239, 239, 239); color: rgb(102, 102, 102);\\" class=\\"nut-avatar nut-avatar-normal nut-avatar-square avatarClass avatarClass--square\\">
  <!--v-if-->
  <!--v-if-->
  <!-- 折叠头像 -->
  <!--v-if-->
</view>"
`;

exports[`should change avatar size when using avatarSize prop: 20px 1`] = `"20px"`;

exports[`should change avatar size when using avatarSize prop: 20px 2`] = `"20px"`;

exports[`should render default slot 1`] = `
"<view class=\\"skeleton\\">
  <!--v-if-->
  <view class=\\"content\\">
    <!--v-if-->
    <view class=\\"blockClass\\" style=\\"width: 100px; height: 100px;\\"></view>
    <view class=\\"content-line\\">
      <view class=\\"title\\"></view>
      <view class=\\"blockClass\\" style=\\"width: 100px; height: 100px;\\"></view>
    </view>
  </view>
</view>"
`;
